Formula/j/julia.rb

```diff
@@ -63,6 +74,12 @@ class Julia < Formula
   end
 
   def install
+    # Compiler selection is not supported for versioned LLVM
+    if DevelopmentTools.clang_build_version == 1600
+      ENV["HOMEBREW_CC"] = Formula["llvm@16"].opt_bin/"clang"
+      ENV["HOMEBREW_CXX"] = Formula["llvm@16"].opt_bin/"clang++"
```

Comment:
Didn't think this worked.

Reply:
Not sure. If we are okay with mixing versions, everything else can compile with system Clang. Just need a different Clang when hitting code in libblastrampoline (but doesn't seem easy to selectively pick for that part, so would need all `CC`)

Reply:
Better to get this fixed upstream so that `USE_SYSTEM_LIBBLASTRAMPOLINE=1` doesn't build `libblastrampoline`.

Reply:
Yeah, doesn't do anything:
```
==> clang -v
Homebrew clang version 16.0.6
Target: arm64-apple-darwin24.0.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm@16/bin
==> clang++ -v
Apple clang version 16.0.0 (clang-1600.0.26.3)
Target: arm64-apple-darwin24.0.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

Reply:
`CXX` doesn't work for `julia` as it runs `which clang++`. Would need to do a PATH modification.

Reply:
> Better to get this fixed upstream so that `USE_SYSTEM_LIBBLASTRAMPOLINE=1` doesn't build `libblastrampoline`.

If newer SuiteSparse can work, then may be possible to avoid as that is what pulls it in
https://github.com/JuliaLang/julia/blob/master/deps/libsuitesparse.mk#L58

Reply:
Don't think it does, unfortunately. But I may be wrong.

Reply:
We could imitate Arch and just ignore the restriction. A bit complicated to patch as part of it is inside a nested tarball.

Official support needs 1.11.0, e.g.
* https://github.com/JuliaLang/julia/commit/c5dcf4f56d9e51ca0a83f5e320377d28e4d4b11a
* https://github.com/JuliaSparse/SparseArrays.jl/commit/f890a1e13fc04c295cec48a9b08b76c6e1a7039a